sources/packages/backend/apps/workers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -333,6 +334,92 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
     return true;
   }
 
+  /**
+   * Updates the last reported assessment for the current assessment.
+   * @param currentAssessmentId current assessment id.
+   * @param applicationId related application id.
+   * @param studyStartDate related study start date.
+   * @param studyEndDate related study end date.
+   * @returns true if the update was executed or false in case the
+   * previousDateChangedReportedAssessment was already present or
+   * in case the update of previousDateChangedReportedAssessment is
+   * not required if the offering start and end dates are the same.
+   */
+  async updateLastReportedAssessment(
+    currentAssessmentId: number,
+    applicationId: number,
+    studyStartDate: string,
+    studyEndDate: string,
```

Comment:
again study start and study end date can come from assessment and not required to be passed to the method.